lib/subproviders/geth_api_double.js

```diff
@@ -212,7 +212,9 @@ GethApiDouble.prototype.eth_getTransactionReceipt = function(hash, callback) {
     var result = null;
 
     if (receipt) {
-      result = receipt.toJSON();
+      if (receipt.block) {
```

Comment:
Combine this `receipt.block` truthiness check with the `if (receipt) {` check above it, i.e., `if (receipt && receipt.block) {`.